src/wrapped_radix_engine_toolkit/high_level/sargon_specific_types/address_union.rs

```diff
@@ -0,0 +1,449 @@
+                        HashSet::<SUT>::from_iter(samples)
+                    );
+                }
+
+                #[test]
+                fn bech32_roundtrip() {
+                    $(
+                        let sut = [< new_ $union_name:snake _sample_ $variant_name:snake _mainnet >]();
+                        let s = [< $union_name:snake _to_string >](&sut);
+                        assert_eq!(
+                            [< new_ $union_name:snake _from_bech32 >](s).unwrap(),
+                            sut
+                        );
+                    )+
+                }
+
+                #[test]
+                fn network_id() {
+                    $(
+                        assert_eq!(
+                            [< $union_name:snake _network_id >](&[< new_ $union_name:snake _sample_ $variant_name:snake _mainnet_other >]()),
+                            NetworkID::Mainnet
+                        );
+                    )+
+
+                    $(
+                        assert_eq!(
+                            [< $union_name:snake _network_id >](&[< new_ $union_name:snake _sample_ $variant_name:snake _stokenet_other >]()),
+                            NetworkID::Stokenet
+                        );
+                    )+
+                }
+
+                #[test]
+                fn map_to_network() {
+
+                    SUT::sample_values_all().into_iter().for_each(|a| {
+                        assert_eq!(
+                            [< $union_name:snake _map_to_network>](&a, NetworkID::Stokenet).network_id(),
+                            NetworkID::Stokenet
+                        );
+                    });
+
+                    SUT::sample_values_all().into_iter().for_each(|a| {
+                        assert_eq!(
+                            [< $union_name:snake _map_to_network>](&a, NetworkID::Mainnet).network_id(),
+                            NetworkID::Mainnet
+                        );
+                    });
+                }
+
+                #[test]
+                fn format_address() {
+                    let test = |a: SUT| {
+                        assert_eq!([< $union_name:snake _formatted>](&a, AddressFormat::Full), a.to_string());
+                        assert_ne!([< $union_name:snake _formatted>](&a, AddressFormat::Default), a.to_string());
+                    };
+                    SUT::sample_values_all().into_iter().for_each(test);
+                }
+
+            }
+
+            $(
+                impl From<$variant_type> for $union_name {
+                    fn from(value: $variant_type) -> Self {
+                        Self::$variant_name(value)
+                    }
+                }
+            )+
+
+            impl IntoScryptoAddress for  $union_name {
+                fn scrypto(&self) -> ScryptoGlobalAddress {
+                    match self {
+                        $(
+                            Self::$variant_name(address) => address.scrypto(),
```

Comment:
pretty nice! simple impl of mapping from the variant to use the method `scrypto()` on it.